### PR TITLE
ROX-12007: v0.4.2 , semantic diff support

### PR DESCRIPTION
**Pull request by shireenf-ibm**
## Description
 * update netpol-analyzer to v0.4.2
   - semantic diff : new `connectivity-diff` command

## Checklist
- [ ] Investigated and inspected CI test results
- [ ] Unit test and regression tests added
- [ ] Evaluated and added CHANGELOG entry if required
- [ ] Determined and documented upgrade steps
- [ ] Documented user facing changes (create PR based on [openshift/openshift-docs](https://github.com/openshift/openshift-docs) and merge into [rhacs-docs](https://github.com/openshift/openshift-docs/tree/rhacs-docs))

If any of these don't apply, please comment below.

## Testing Performed:
unit tests + bats tests

In addition to reviewing your code, reviewers **must** also review your testing
instructions and make sure they are sufficient.

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -70,7 +70,7 @@ require (
 	github.com/mitchellh/hashstructure/v2 v2.0.2
 	github.com/moby/sys/mountinfo v0.6.2
 	github.com/np-guard/cluster-topology-analyzer v1.7.0
-	github.com/np-guard/netpol-analyzer v0.2.2
+	github.com/np-guard/netpol-analyzer v0.4.2
```
Comment (janisz) [inline go.mod:73]: How about splitting this PR into update and new command?

**Comment by vikin91** [comment]
This is too big to review for now, need to wait for #7173 to be merged or splitting into two as @janisz suggested.

File: go.mod (hunk 1 above)
Comment (adisos) [reply go.mod:73]: closing this PR, will split to two separate PRs.

**Pull request closed without merging**